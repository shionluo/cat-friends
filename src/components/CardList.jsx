import React, { useEffect } from "react";

import { connect } from "react-redux";

//-- Actions --//
import { requestCats } from "../action";

//-- Components --//
import Card from "./Card";

//----------------------------------------------------//
//----------------------------------------------------//

const mapStateToProps = state => ({
  cats: state.requestCats.cats,
  isPending: state.requestCats.isPending,
  searchField: state.searchCats.searchField
});

const mapDispatchToProps = dispatch => ({
  requestCats: () => dispatch(requestCats())
});

const CardList = ({ cats, isPending, searchField, requestCats }) => {
  useEffect(() => {
    requestCats();
  }, [requestCats]);

  const filterCats = cats.filter(cat =>
    cat.name.toLowerCase().includes(searchField.toLowerCase())
  );

  return isPending ? (
    <h1 className="tc">Loading ...</h1>
  ) : (
    <div>
      {filterCats.map(cat => (
        <Card key={cat.id} {...cat} />
      ))}
    </div>
  );
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(CardList);
