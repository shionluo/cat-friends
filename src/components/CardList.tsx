import React, { useEffect } from "react";

import { connect } from "react-redux";

//-- Actions --//
import { requestCats } from "../actions/action";

//-- Components --//
import Card from "./Card";

//-- Type --//
import { AppState } from "../store/configureStore";
import { ThunkDispatch } from "redux-thunk";

//----------------------------------------------------//
//----------------------------------------------------//

const mapStateToProps = (state: AppState) => ({
  cats: state.requestCats.cats,
  isPending: state.requestCats.isPending,
  searchField: state.searchCats.searchField
});

const mapDispatchToProps = (dispatch: ThunkDispatch<any, any, any>) => ({
  requestCats: () => dispatch(requestCats())
});

interface Props {
  cats: any[]
  isPending: boolean;
  searchField: string;
  requestCats: () => void;
}

const CardList: React.SFC<Props> = ({
  cats,
  isPending,
  searchField,
  requestCats
}) => {
  useEffect(() => {
    requestCats();
  }, [requestCats]);

  const filterCats = cats.filter(cat =>
    cat.name.toLowerCase().includes(searchField.toLowerCase())
  );

  return isPending ? (
    <h1 className="tc">Loading ...</h1>
  ) : (
    <div>
      {filterCats.map(cat => (
        <Card key={cat.id} {...cat} />
      ))}
    </div>
  );
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(CardList);
