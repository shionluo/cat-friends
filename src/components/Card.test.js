import React from "react";

//-- Enzyme --//
import { shallow } from "enzyme";
import "../setupTests";

//-- Components --//
import CardList from "./Card";

//----------------------------------------------------//  
//----------------------------------------------------//

it("Expect to render Card Component", () => {
  expect(shallow(<CardList />).debug()).toMatchSnapshot();
});