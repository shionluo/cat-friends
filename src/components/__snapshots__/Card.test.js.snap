// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect to render Card Component 1`] = `
"<div className=\\"tc bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5\\">
  <img src=\\"https://robohash.org/undefined?set=set4&size=250x250\\" alt=\\"cat\\" />
  <div>
    <h2 />
    <p />
  </div>
</div>"
`;
